packages/ai-codex/src/common/codex-preferences.ts

```diff
@@ -0,0 +1,50 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { AI_CORE_PREFERENCES_TITLE } from '@theia/ai-core/lib/common/ai-core-preferences';
+import { nls, PreferenceSchema } from '@theia/core';
+
+export const CODEX_API_KEY_PREF = 'ai-features.codex.apiKey';
+export const CODEX_SDK_PATH_PREF = 'ai-features.codex.sdkPath';
+export const CODEX_SANDBOX_MODE_PREF = 'ai-features.codex.sandboxMode';
+
+export const CodexPreferencesSchema: PreferenceSchema = {
+    properties: {
+        [CODEX_API_KEY_PREF]: {
+            type: 'string',
+            markdownDescription: nls.localize('theia/ai/codex/apiKey/description',
+                'OpenAI API key for Codex. If not set, falls back to the shared OpenAI API key (`ai-features.openAiOfficial.openAiApiKey`). Can also be set via `OPENAI_API_KEY` environment variable.'),
+            title: AI_CORE_PREFERENCES_TITLE,
+        },
+        [CODEX_SDK_PATH_PREF]: {
+            type: 'string',
+            markdownDescription: nls.localize('theia/ai/codex/sdkPath/description',
+                'Custom path to Codex SDK installation. If not specified, the system will attempt to resolve the path automatically from the global npm installation.'),
+            title: AI_CORE_PREFERENCES_TITLE,
+        },
+        [CODEX_SANDBOX_MODE_PREF]: {
```

Comment:
For now this can probably stay a preference, but mid-term this should really be a session-setting instead of a user setting. But we don't have a UI or generic support for modes yet, so for now we can leave it like this, or we just hard-code workspace-write for now in the agent, and update it so support the other two modes once we have mode support in Theia AI.